src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -27,22 +29,30 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/login";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    boolean loginStatus = userService.isUserLoggedIn();
+    String urlToRedirect = "/index.html";
 
-      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    ArrayList<String> userDetails = new ArrayList();
+    if (loginStatus) {
+      userDetails.add("true");
```

Comment:
Adding "true" or "false" is a bit surprising since it seems that this arraylist is for holding html. Perhaps a comment indicating that this is for testing purposes would be appropriate (if that's what it's for).

Reply:
Ah nvm, just checked how you're using this below. This should definitely be sent as a json object, not as an array. It's much more error prone to try to remember which index is which value then to have named keys : )

Reply:
makes sense, implemented it as a userDetails property!